docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,128 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+User Agent Filter module allows you to allow or deny traffic based on the user-agent request header. You can define allow and deny regular expression rules that apply to the incoming user-agent request header.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### SSH
+
+```bash
+ssh -R 443:localhost:80 connect.ngrok-agent.com http \
+ --ua-filter-allow="(foo/(\d)+.(\d)+)" \
+ --ua-filter-deny="(bar/(\d)+.(\d)+)" \
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+ example:
+   proto: https
+   addr: 80
+   ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+   ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
+
+```go
+import (
+	"context"
+	"net"
+
+	"golang.ngrok.com/ngrok"
+	"golang.ngrok.com/ngrok/config"
+)
+
+func listenUserAgentFIlter(ctx context.Context) net.Listener {
+	listener, _ := ngrok.Listen(ctx,
+		config.HTTPEndpoint(
+			config.WithUserAgentFilter( []string{"(foo)/(\d)+.(\d)+"}, []string{"bar/(\d)+.(\d)+"}),
+		),
+		ngrok.WithAuthtokenFromEnv(),
+	)
+	return listener
+}
+```
+
+## Behavior
+
+### Rule Evaluation
+
+On request, the User Agent Filter module will check the incoming `user-agent` header value against the list of defined allow and deny regular expression rules.
+
+Requests with no `user-agent` header specified will still be matched against the list of defined rules. If you wish to reject these you can add a deny rule for empty strings: `^$`
+
+Requests with multiple `user-agent` headers will be matched as a single value.
```

Comment:
what does this mean "will be matched as a single value"?

Reply:
Let say we send this request 
```
curl --location https://mo.ngrok.io.lan \                                                                                                                                                                                                 
--header 'Content-Type: text/plain' \
--header 'User-Agent: foo/1.1' \
--header 'User-Agent: bar/1.1' \
--header 'User-Agent: buz/1.1' \
--data 'https://www.youtube.com/watch?v=djyTG19Achg' -k -v
```
with a tunnel configured with
```
--ua-filter-allow="(foo/(\d)+.(\d)+)"
--ua-filter-deny="(bar/(\d)+.(\d)+)"
```
It will only evaluate the first user agent header